project.py

```diff
@@ -487,6 +530,207 @@ def preprocesing(self, image):
+                                               (tmp.shape[1], tmp.shape[0]))
+                elif occlusion_mask.shape < tmp.shape:
+                    tmp = cv.resize(tmp, (occlusion_mask.shape[1],
+                                          occlusion_mask.shape[0]))
+                intersection = cv.bitwise_and(occlusion_mask, tmp)
+                count = cv.countNonZero(intersection)
+                if self.stored_section[x][y][0] is None:
+                    self.stored_section[x][y][0] = tmpimg
+                    self.stored_section[x][y][1] = 0
+                if count <= 0:
+                    self.stored_section[x][y][1] += 1
+                    if self.debug:
+                        color = (0, 255, 0)
+                else:
+                    if self.debug:
+                        color = (0, 0, 255)
+
+                if self.debug:
+                    cv.rectangle(self.debug_image, (x*section_size[0],
+                                                    y*section_size[1]),
+                                                   ((x+1)*section_size[0],
+                                                    (y+1)*section_size[1]),
+                                 color, 1)
+
+    def stitch_board(self, board):
+        """
+        Function takes sections of a board and based on given thresholding
+        values tries to stitch them into one final image. If we seen
+        section more than given section thresholding value we sow this
+        section into final slide. If section is lower than section reject
+        threshold we sow last good section into final image. If value is
+        between these two thresholds we check how similar is this section
+        to last good save one if they are similar last good one is sow if they
+        are not we got some new information and new section is put into final
+        slide
+
+        Args:
+            board(numpy.ndarray): Image of a board
+
+        Returns:
+            (numpy.ndarray): Final image of a board from stitched sections
+        """
+        height, width = board.shape[:2]
+        section_size = (width/self.grid_size[0], height/self.grid_size[1])
+        if self.last_saved_slide is None:
+            blank = np.zeros((height, width, 3), dtype="uint8")
+        else:
+            blank = self.last_saved_slide.copy()
+        if self.debug:
+            font_offset = 35
+        for x in range(0, self.grid_size[0]):
+            for y in range(0, self.grid_size[1]):
+                seen = self.stored_section[x][y][1]
+                section = self.stored_section[x][y][0]
+                boundaries = self.__get_section_boundary__(x, y, section_size,
+                                                           width, height)
+                if self.last_saved_section[x][y][0] is not None:
+                    last_good_section = self.last_saved_section[x][y][0]
+                else:
+                    blank[boundaries[0]:boundaries[1],
```

Comment:
@mrshu I'm sorry but I don't know what else to do with this. I could probably rename it to not be boundaries[0] but something like section_x0 or x0 or boundary_x0 so it would be probably more clearer what it does.